src/main/java/io/confluent/connect/jdbc/dialect/SqlServerDatabaseDialect.java

```diff
@@ -324,7 +325,10 @@ public String buildUpsertQueryStatement(
     ExpressionBuilder builder = expressionBuilder();
     builder.append("merge into ");
     builder.append(table);
-    builder.append(" with (HOLDLOCK) AS target using (select ");
+    if (this.config.getBoolean(JdbcSinkConfig.MSSQL_USE_MERGE_HOLDLOCK)) {
```

Comment:
```suggestion
    if (this.config.useHoldlockInMerge) {
```

Reply:
Needs casting to JdbcSinkConfig

((JdbcSinkConfig) this.config).useHoldlockInMerge